datafusion/optimizer/src/simplify_expressions/regex.rs

```diff
@@ -131,6 +131,15 @@ fn str_from_literal(l: &Literal) -> Option<&str> {
     }
 }
 
+/// returns a str represented by `Literal` if it contains a valid utf8
+/// sequence and is unsafe for like (maybe has  '%' or '_')
+fn str_from_literal_unsafe_like(l: &Literal) -> Option<&str> {
```

Comment:
I was somewhat confused about this name and the comments as the code doesn't actually check for '%' or '_`

What would you think about renaming it something like

```suggestion
/// returns a str represented by `Literal` if it contains a valid utf8
fn str_from_literal(l: &Literal) -> Option<&str> {
```

And then renaming the existing `str_from_literal` to something like this:

to 

```rust
/// returns a str represented by `Literal` if it contains a valid utf8
/// sequence and is safe for `like` (has no '%' and '_')
fn like_str_from_literal(l: &Literal) -> Option<&str> {
```